### PR TITLE
Added the body of TableService AddPrByUsernameAsync

### PR DESCRIPTION
**Pull request by CrypticEngima**

### CODE REVIEW

File: HacktoberfestProject.Web/Services/TableService.cs [hunk 1]
```diff
@@ -18,20 +19,41 @@ public TableService(IUserRepository userRepository)
...
 
-        public Task<ServiceResponse<IEnumerable<Pr>>> AddPrByUsernameAsync(string username, Pr pr)
+        public async Task<ServiceResponse<Pr>> AddPrByUsernameAsync(string username, string owner, string repositoryName, Pr pr)
         {
-            throw new NotImplementedException();
+            var user = await GetUser(username);
+
+            Repository repository = GetReporitory(user , owner, repositoryName);
+
+            var serviceResponse = new ServiceResponse<Pr>
```
Comment (Layla-P) [inline HacktoberfestProject.Web/Services/TableService.cs:28]: We could create a constructor for this to make it easier in the future

File: HacktoberfestProject.Web/Services/TableService.cs [hunk 2]
```diff
@@ -18,20 +19,41 @@ public TableService(IUserRepository userRepository)
...
+            {
+                ServiceResponseStatus = ServiceResponseStatus.Created
+            };
+
+            if (!repository.Prs.All(p => p.PrId == pr.PrId))
```
Comment (Layla-P) [inline HacktoberfestProject.Web/Services/TableService.cs:33]: PrId is not a unique identifier as far as I am aware. We would need to check against owner, repository _and_ id I believe.

File: HacktoberfestProject.Web/Services/TableService.cs [hunk 3]
```diff
@@ -18,20 +19,41 @@ public TableService(IUserRepository userRepository)
...
+            {
+                repository.Prs.Add(pr);
+                user.RepositoryPrAddedTo.Add(repository);
+                await _userRepository.UpdateAsync(user);
+                serviceResponse.Message = "Pr Added!";
+            }
+            else
+            {
+                serviceResponse.Message = "Pr was already added!";
+                serviceResponse.ServiceResponseStatus = ServiceResponseStatus.Ok;
```
Comment (Layla-P) [inline HacktoberfestProject.Web/Services/TableService.cs:43]: We should add another responseEnum to reflect this outcome

File: HacktoberfestProject.Web/Services/TableService.cs [hunk 4]
```diff
@@ -18,20 +19,41 @@ public TableService(IUserRepository userRepository)
...
         }
-        
+
         public async Task<ServiceResponse<IEnumerable<Pr>>> GetPrsByUsernameAsync(string username)
         {
-            var user = new User(username);
-
-            var userEntity = await _userRepository
-                                        .ReadAsync(user);
+            var user = await GetUser(username);
```
Comment (Layla-P) [inline HacktoberfestProject.Web/Services/TableService.cs:53]: GetUserAsync?

File: HacktoberfestProject.Web/Services/TableService.cs [hunk 5]
```diff
@@ -66,11 +87,38 @@ public async Task<ServiceResponse<User>> GetUserByUsernameAsync(string username)
...
+        {
+            User user = new User(username);
+
+            User tablestorageUser = await _userRepository.ReadAsync(user);
+
+            return tablestorageUser;
+
+        }
+
+        private Repository GetReporitory(User user, string owner, string repositoryName)
```
Comment (Layla-P) [inline HacktoberfestProject.Web/Services/TableService.cs:107]: Typo: GetRespository

File: HacktoberfestProject.Web/Services/TableService.cs (hunk 2 above)
Comment (CrypticEngima) [reply HacktoberfestProject.Web/Services/TableService.cs:33]: The current check you are seeing here is based on the repository that has already been filtered to owner and repository name.

File: HacktoberfestProject.Web/Services/TableService.cs (hunk 3 above)
Comment (CrypticEngima) [reply HacktoberfestProject.Web/Services/TableService.cs:43]: understood and will implament

File: HacktoberfestProject.Web/Services/TableService.cs (hunk 4 above)
Comment (CrypticEngima) [reply HacktoberfestProject.Web/Services/TableService.cs:53]: my bad

File: HacktoberfestProject.Web/Services/TableService.cs (hunk 5 above)
Comment (CrypticEngima) [reply HacktoberfestProject.Web/Services/TableService.cs:107]: sorry mistyped lol i'll fix now

**Review by Layla-P — APPROVED** [review]
Looking good, thank you for updating :-)

**Pull request merged**